src/lib/TasksManager.ts

```diff
@@ -81,67 +102,97 @@ export class TasksManager {
 
     try {
       const page = new BrowserPage();
-      await page.create(this.#browser);
+
+      // It seems page creation can hang infinitely in puppeteer
+      // so we want it fail as soon as possible to retry on an other pod
+      await Promise.race([
+        page.create(this.#browser),
+        (async (): Promise<void> => {
```

Comment:
Care, it doesn't work, `Promise.race` don't stop the other promises and it eventually fulfills, stopping the TaskManager.
The promise needs to be manually aborted, or we can use a `pageCreated = true` flag.

Reply:
mmh my bad, I thought so. What's weird is that it does not break the test :D

Reply:
Probably because it does the requests all at once, and the TaskManager gracefully waits for everything to be finished before stopping the browser.